File: problem1/solution.cpp
#include <string>
#include <vector>
#include <stack>

using namespace std;

vector<int> solution(vector<int> prices) {
    vector<int> answer(prices.size(), 0);
    stack<int> records;
    
    for (int i = 0; i < prices.size(); i++) {
        while (!records.empty() && prices[records.top()] > prices[i]) {
            answer[records.top()] = i - records.top();
            records.pop();
        }
        records.push(i);
    }
    
    while (!records.empty()) {
        answer[records.top()] = prices.size() - records.top() - 1;
        records.pop();
    }
    
    return answer;
}

File: problem1/solution.js
const solution = (prices) => {
	let answer = Array(prices.length);
	let records = [];

	prices.forEach((price, i) => {
		while (records.length !== 0 && prices[records[records.length - 1]] > price) {
			answer[records[records.length - 1]] = i - records[records.length - 1];
			records.pop();
		}
		records.push(i);
	});

	while (records.length !== 0) {
		answer[records[records.length - 1]] = prices.length - records[records.length - 1] - 1;
		records.pop();
	}

	return answer;
}